test/evp_kdf_test.c

```diff
@@ -1651,6 +1651,30 @@ OSSL_PARAM *construct_snmpkdf_params(char *digest, unsigned char *eid, size_t ei
     return p_ret;
 }
 
+static int test_kdf_snmpkdf_null_output_buf(void)
+{
+    int ret;
+    EVP_KDF_CTX *kctx = NULL;
+    OSSL_PARAM *params = NULL;
+    unsigned char out[SHA_DIGEST_LENGTH];
+
+    static unsigned char EID1[] = {
+          0x00, 0x00, 0x02, 0xb8, 0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
+    static unsigned char password1[] = {'L', 'P', 'U', 'Q', 'D', 'L', 's', 'K' };
+
+    params = construct_snmpkdf_params("sha1", EID1, sizeof(EID1),
+                                      (unsigned char *)password1, 1048577);
+
+    /* Negative test - derive should fail */
+    ret = TEST_ptr(params)
+        && TEST_ptr(kctx = get_kdfbyname(OSSL_KDF_NAME_SNMPKDF))
+        && TEST_int_eq(EVP_KDF_derive(kctx, NULL, sizeof(out), NULL), 0);
```

Comment:
I am wondering if the check for a NULL key should be moved up to the EVP_KDF_derive().. Does it make sense for any KDF to allow this to be NULL?
I can see existing KDF's that are not checking this at the lower level so they would crash currently.

Reply:
I don't see a reason for that. Moving the check into EVP_KDF_derive would certainly catch it earlier and cover all KDFs.